plugin/testdata/patched_url_test.lua

```diff
@@ -0,0 +1,16 @@
+local typedefs = require "kong.db.schema.typedefs"
+local url = require "socket.url"
+
+return {
+    name = "patched-url-test",
+    fields = {
+        { consumer = typedefs.no_consumer },
+        { protocols = typedefs.protocols_http },
+        { config = {
+            type = "record",
+            fields = {
+                { uuid = typedefs.uuid },
+            },
+        },
+        },
+    },}
```

Comment:
This schema needs more for the test; `url` is never used so there is no validation happening here. The current field `uuid` has no value for `url`. Create a custom_validator function to verify that `url.parse(<a-url-field>)` is working as expected:

```
<url> ::= <scheme>://<authority>/<path>;<params>?<query>#<fragment>
<authority> ::= <userinfo>@<host>:<port>
<userinfo> ::= <user>[:<password>]
<path> :: = {<segment>/}<segment>
```

You will want to check:
- scheme
- authority
- userinfo
- user
- password
- host
- port
- path
- params
- query
- fragment

This test could also be extended to validated `patched.url` as well in the validator that way we are ensuring backwards compatibility through a regression.